zookeeper-server/src/main/java/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -782,6 +782,10 @@ void queueCallback(AsyncCallback cb, int rc, String path, Object ctx) {
         eventThread.queueCallback(cb, rc, path, ctx);
     }
 
+    protected void onConnecting(InetSocketAddress addr) {
+        LOG.info("Try to connect to server {}.", addr);
```

Comment:
Since the primary reason of adding this is for injection in unit tests, I'd suggest we mark this method which a comment like "for tests only", and remove the log statement which if leaving as is, will create a lot of noises in client side logs. We can still override an empty method in test implementation.

Reply:
Thanks your comments. Accept it. I will add comments to declare it as "test only" and leave this method as empty.